examples/CMakeLists.txt

```diff
@@ -46,17 +46,39 @@ IF(DEAL_II_COMPONENT_EXAMPLES)
     PATTERN "DTMB-5415_bulbous_bow.iges"  # step-54
     )
 
-  MESSAGE(STATUS "Setting up examples - Done")
-ENDIF()
+  #
+  # Set up all executables:
+  #
+  FILE(GLOB _steps ${CMAKE_CURRENT_SOURCE_DIR}/step-*/*.cc)
+  FOREACH(_step ${_steps})
+    GET_FILENAME_COMPONENT(_name ${_step} NAME_WE)
+    GET_FILENAME_COMPONENT(_directory ${_step} DIRECTORY)
+    SET(_set_up TRUE)
+    IF(EXISTS "${_directory}/dependency")
+      FILE(READ "${_directory}/dependency" _dependency)
+      # I will so totally go to hell for that:
+      RUN_COMMAND("
+        IF(NOT (${_dependency}))
+          MESSAGE(STATUS \"  ${_name} - dependency not satisfied\")
+          SET(_set_up FALSE)
+        ENDIF()"
```

Comment:
So could we, instead of a list of dependency variables, put a little CMake script into `step-XX/doc/dependencies`? We could then just include *that*.